ukrdc_sqla/ukrdc.py

```diff
@@ -494,6 +529,37 @@ class Allergy(Base):
     id = Column(String, primary_key=True)
     pid = Column(String, ForeignKey("patientrecord.pid"))
 
+    creation_date = Column(DateTime, nullable=False, server_default=text("now()"))
+    idx = Column(Integer)
+
+    allergycode = Column(String(100))
```

Comment:
Does this need Synonyms as elsewhere to make it match with https://github.com/renalreg/ukrdc_database/blob/master/ukrdc/database/models/ukrdc.py#L579 ?

Reply:
I mean I'd like to eventually see all the synonyms removed so we're using the normal column names everywhere, but I appreciate that's not worth the disruption at this stage, and can add them here if you think there's value.

Reply:
Are the symonyms being used anywhere already?

If not then I'd say don't add new ones (and indeed remove the rest?) and assume that to change which library you use will require some attribute changes. Otherwise I think it makes sense to add them to be consistent.

Reply:
Basically this library was based on the ukrdc database models created ages ago which include all the pyxb stuff. Those had attribute names that weren't identical to the column names (things like extra underscores).

In order to keep some consistency, I added all fields to this library under their actual column names, and created synonyms for compatibility with anything using those "imported" field names.

I see 3 totally fine options here, so pick your fav:

1. We keep synonyms all over the place, and add them to these new models and leave them all in indefinitely
2. We keep _existing_ synonyms for backwards compatibility, but avoid creating any new ones
3. We do a major library release that removes all synonyms. Since it would increment a major version number, you can always keep the old version in your dependencies for compatibility, but we encourage everyone to update their code to use the real column names.

My preference would be 2 or ideally 3, but tbh I'm not overly passionate about what we do here.

Reply:
I don't really mind other than that I'd like whatever makes getting rid of the duplication in ukrdc_database easiest.

The other question is whether if we think the DDL is "wrong" whether that also needs changing. I'd rather not, but I'd rather do it now than later.

Reply:
Ok I'll leave synonyms in as that will make replacing the ukrdc_database models easiest by miles, as it preserves backwards compatibility